iam.tf

```diff
@@ -1,3 +1,9 @@
+locals {
+  iam_service_types = ["bus", "event_api", "sfn"]
+  target_types      = [for k, v in var.targets : k if length(v) > 0]
+  needs_iam         = length(setintersection(local.iam_service_types, local.target_types)) > 0
```

Comment:
What did it do when it tried creating roles for other types?

Reply:
@ianbender 
```
╷
│ Error: updating EventBridge Target (highwing-trigger-hartford-component-terraform-20230915155952202600000007): ValidationException: RoleArn is not supported for target arn:aws:sqs:us-east-1:291822390326:submissions-requests.
│       status code: 400, request id: b9afd4b1-ddae-4a78-b8b7-3cb30519756b
│ 
│   with module.hartford_component.module.start_submission.aws_cloudwatch_event_target.event_target["start-submission"],
│   on .terraform/modules/hartford_component.start_submission/main.tf line 36, in resource "aws_cloudwatch_event_target" "event_target":
│   36: resource "aws_cloudwatch_event_target" "event_target" {
```

What's infuriating is [the docs](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule.html) only state:
> For many target types, EventBridge needs permissions to send events to the target. In these cases, EventBridge can create the IAM role needed for your rule to run. 

I can't find any way to find the list of types that _do_ require it besides just poking at the console UI

Reply:
Oh yeah, that's super annoying. I guess you can kind of infer them from the work in [this module](https://github.com/terraform-aws-modules/terraform-aws-eventbridge/blob/master/iam.tf).

Reply:
Hmm, is it possible that you need to pass the role name in instead of the ARN? I know there are a few resources that are finicky with that.